File: CheckIfSorted.py
def arraySortedOrNot(arr, n):
    # Nếu mảng có một hoặc không có phần tử, mảng đã được sắp xếp
    if (n == 0 or n == 1):
        return True

    # Duyệt qua mảng để tìm cặp phần tử không sắp xếp
    for i in range(1, n):
        # Nếu tìm thấy cặp phần tử mà phần tử trước lớn hơn phần tử sau
        if (arr[i-1] > arr[i]):
            return False

    # Nếu không tìm thấy cặp phần tử nào sai
    return True


# Nhập mảng từ người dùng
arr = list(map(int, input("Hãy nhập các phần tử của mảng cách nhau bởi khoảng trắng: ").split()))

# Tính toán số lượng phần tử trong mảng
n = len(arr)

# Kiểm tra mảng có sắp xếp không và in kết quả
if (arraySortedOrNot(arr, n)):
    print("Yes")
else:
    print("No")

File: CheckIfSorted.c
#include <stdio.h>
#include <stdbool.h>
bool isSorted(int arr[], int n) {
    // Duyệt qua mảng từ đầu đến trước khi đến phần tử cuối cùng
    for (int i = 0; i < n - 1; i++) {
        if (arr[i] > arr[i + 1]) {
            return false;
        }
    }
    return true;
}
int main() {
    int n;
    printf("Nhập số lượng phần tử trong mảng: ");
    scanf("%d", &n);
    
    int arr[n];
    printf("Nhập các phần tử của mảng cách nhau bởi khoảng trắng: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    if (isSorted(arr, n)) {
        printf("Mảng đã được xếp theo thứ tự tăng dần.\n");
    } else {
        printf("Mảng chưa được xếp theo thứ tự tăng dần.\n");
    }
    
    return 0;
}
